### PR TITLE
Emit cumulative counters instead of gauges (NerveUWSGI)

### PR DESCRIPTION
**Pull request by dichiarafrancesco**
We want to change the type emitted for "meters" and "timers" metrics with "rollup" == "count" from Gauges to Cumulative Counters.

Old output example:
[
{pyramid_uwsgi_metrics.tweens.4xx-responses gauge 7936 map[rollup:count]} 
{pyramid_uwsgi_metrics.tweens.4xx-responses gauge 0.19989366873361622 map[rollup:mean_rate]}
{pyramid_uwsgi_metrics.tweens.4xx-responses gauge 0.19700330306194488 map[rollup:m15_rate]}
{pyramid_uwsgi_metrics.tweens.4xx-responses gauge 0.19102634991044923 map[rollup:m5_rate]}
{pyramid_uwsgi_metrics.tweens.4xx-responses gauge 0.15404916698203022 map[rollup:m1_rate]}
]

New output ex:
[
{pyramid_uwsgi_metrics.tweens.4xx-responses.count cumcounter 542768 map[rollup:count]}
{pyramid_uwsgi_metrics.tweens.4xx-responses gauge 0.20030535922868342 map[rollup:mean_rate]}
{pyramid_uwsgi_metrics.tweens.4xx-responses gauge 0.2003738607402713 map[rollup:m15_rate]}
{pyramid_uwsgi_metrics.tweens.4xx-responses gauge 0.20121278046362234 map[rollup:m5_rate]}
{pyramid_uwsgi_metrics.tweens.4xx-responses gauge 0.19881931795422336 map[rollup:m1_rate]}
]

This change is backward compatible, so don't worry! To enable this change you need to modify the NerveUWGSI config make it include {"enableCumulativeCounters": true}

### CODE REVIEW

**Comment by gnufied** [comment]
This will also change metric type etc for Java services. Nothing wrong with that itself, but as discussed we should do this step-by-step, I think.

File: src/fullerite/collector/nerve_uwsgi.go [hunk 1]
```diff
@@ -98,12 +101,18 @@ func newNerveUWSGI(channel chan metric.Metric, initialInterval int, log *l.Entry
...
-type nerveUWSGICollector struct {
-	baseCollector
+func (n *nerveUWSGICollector) Configure(configMap map[string]interface{}) {
+	if val, exists := configMap["queryPath"]; exists {
+		n.queryPath = val.(string)
+	}
+	if val, exists := configMap["configFilePath"]; exists {
+		n.configFilePath = val.(string)
+	}
+	if val, exists := configMap["servicesWhitelistCCount"]; exists {
```
Comment (gnufied) [inline src/fullerite/collector/nerve_uwsgi.go]: typo? In any case -  I think we should simply use the string `whitelist` as the key here because people often assume it to be that

File: src/fullerite/collector/nerve_uwsgi.go [hunk 2]
```diff
@@ -427,6 +344,109 @@ func parseNestedMetricMaps(
...
+//    "mean_rate": 100
+// }
+// and metricname and metrictype and returns metrics for each name:rollup pair
+func metricFromMap(metricMap map[string]interface{}, metricName string, metricType string, cumulCounterEnabled bool) []metric.Metric {
+	results := []metric.Metric{}
+
+	for rollup, value := range metricMap {
+		mName := metricName
+		mType := metricType
+		matched, _ := regexp.MatchString("m[0-9]+_rate", rollup)
```
Comment (gnufied) [inline src/fullerite/collector/nerve_uwsgi.go]: Will it not be cleaner to emit the metric here and then blacklist it? I know - we will have to implement support for blacklisting of metrics at collector level, but it may be worth doing that?

**Comment by gnufied** [comment]
fix & ship!

File: src/fullerite/collector/nerve_uwsgi.go (hunk 2 above)
Comment (dichiarafrancesco) [inline src/fullerite/collector/nerve_uwsgi.go]: I think you're right. we should implement blacklisting as diamond does at collector level. But i really think it's a different task. let's do this in another PR.

File: src/fullerite/collector/nerve_uwsgi.go (hunk 1 above)
Comment (dichiarafrancesco) [inline src/fullerite/collector/nerve_uwsgi.go]: it's a slice. and i think should be like that.

Comment (gnufied) [inline src/fullerite/collector/nerve_uwsgi.go]: nooo mate - but we are calling the whitelist `servicesWhitelistCCount`! what is that `CCount` for? We already saw side effect of having complicated configuration keys when people in their head assume it just to be the string `whitelist`. 

I would rather do the simplest thing and call it `whitelist`

File: src/fullerite/collector/nerve_uwsgi.go [hunk 3]
```diff
@@ -427,6 +344,109 @@ func parseNestedMetricMaps(
...
+		}
+	}
+
+	// if nothing else works try for rate
+	return collectRate(jsonMap, metricName)
+}
+
+// serviceInWhitelist returns true if the service name passed as argument
+// is found among the ones whitelisted by the user
+func (n *nerveUWSGICollector) serviceInWhitelist(service string) bool {
```
Comment (gnufied) [inline src/fullerite/collector/nerve_uwsgi.go]: I am thinking - if we built a dictionary from supplied whitelist - this check can be very much simplified and will be probably quicker too.

File: src/fullerite/collector/nerve_uwsgi.go (hunk 1 above)
Comment (dichiarafrancesco) [inline src/fullerite/collector/nerve_uwsgi.go]: ah you mean. the name. go for `whitelist`. my bad.

File: src/fullerite/collector/nerve_uwsgi.go (hunk 3 above)
Comment (dichiarafrancesco) [inline src/fullerite/collector/nerve_uwsgi.go]: i don't think this iteration has huge impact on performance.

Comment (gnufied) [inline src/fullerite/collector/nerve_uwsgi.go]: Yeah it is not about performance per-se, but I am thinking it will be cleaner implementation too.

File: src/fullerite/collector/nerve_uwsgi.go (hunk 1 above)
Comment (dichiarafrancesco) [inline src/fullerite/collector/nerve_uwsgi.go]: looking better i discovered that fullerite already has a `collectorWhitelist` config value, so using just   `whitelist` could be misleading. so i think we need to change it back to `servicesWhitelist`

Comment (gnufied) [inline src/fullerite/collector/nerve_uwsgi.go]: okay

**Comment by gnufied** [comment]
:shipit:
